clickhouse-jdbc/src/main/java/com/clickhouse/jdbc/ResultSetImpl.java

```diff
@@ -1,981 +1,1236 @@
+        checkClosed();
+        return reader.getString(columnIndex);
     }
 
     @Override
     public boolean getBoolean(int columnIndex) throws SQLException {
-        return false;
+        checkClosed();
+        return reader.getBoolean(columnIndex);
     }
 
     @Override
     public byte getByte(int columnIndex) throws SQLException {
-        return 0;
+        checkClosed();
+        return reader.getByte(columnIndex);
     }
 
     @Override
     public short getShort(int columnIndex) throws SQLException {
-        return 0;
+        checkClosed();
+        return reader.getShort(columnIndex);
     }
 
     @Override
     public int getInt(int columnIndex) throws SQLException {
-        return 0;
+        checkClosed();
+        return reader.getInteger(columnIndex);
     }
 
     @Override
     public long getLong(int columnIndex) throws SQLException {
-        return 0;
+        checkClosed();
+        return reader.getLong(columnIndex);
     }
 
     @Override
     public float getFloat(int columnIndex) throws SQLException {
-        return 0;
+        checkClosed();
+        return reader.getFloat(columnIndex);
     }
 
     @Override
     public double getDouble(int columnIndex) throws SQLException {
-        return 0;
+        checkClosed();
+        return reader.getDouble(columnIndex);
     }
 
     @Override
     public BigDecimal getBigDecimal(int columnIndex, int scale) throws SQLException {
-        return null;
+        checkClosed();
+        return reader.getBigDecimal(columnIndex);
     }
 
     @Override
     public byte[] getBytes(int columnIndex) throws SQLException {
-        return new byte[0];
+        checkClosed();
+        return reader.getByteArray(columnIndex);
     }
 
     @Override
     public Date getDate(int columnIndex) throws SQLException {
-        return null;
+        checkClosed();
+        LocalDate localDate = reader.getLocalDate(columnIndex);
+        if(localDate == null) {
+            return null;
+        }
+        return Date.valueOf(localDate);
     }
 
     @Override
     public Time getTime(int columnIndex) throws SQLException {
-        return null;
+        checkClosed();
+        LocalDateTime localDateTime = reader.getLocalDateTime(columnIndex);
+        if(localDateTime == null) {
+            return null;
+        }
+        return Time.valueOf(localDateTime.toLocalTime());
     }
 
     @Override
     public Timestamp getTimestamp(int columnIndex) throws SQLException {
-        return null;
+        checkClosed();
+        LocalDateTime localDateTime = reader.getLocalDateTime(columnIndex);
+        if(localDateTime == null) {
+            return null;
+        }
+        return Timestamp.valueOf(localDateTime);
     }
 
     @Override
     public InputStream getAsciiStream(int columnIndex) throws SQLException {
-        return null;
+        checkClosed();
+        return new ByteArrayInputStream(reader.getString(columnIndex).getBytes(StandardCharsets.UTF_8));
```

Comment:
I think, this also should be part of the reader. Please add a TODO: 

Here is some info about stream content -https://docs.oracle.com/cd/A97335_01/apps.102/a83724/oralob2.htm#1043220
```
To read from a CLOB, use the getAsciiStream() or getCharacterStream() method of an oracle.sql.CLOB object to retrieve the entire CLOB as an input stream. The getAsciiStream() method returns an ASCII input stream in a java.io.InputStream object. The getCharacterStream() method returns a Unicode input stream in a java.io.Reader object.
```
So it is not correct to return UTF8, I think. 

lets just throw exception that it is not implemented. Will sort it out later.